File: learning_cpp/exercices_from_claude_delannoy/chap_2/harmonic_sum.py
import time
n = pow(10,8)
harmonic_sum = 1
begin_time = time.time()
for i in range(2,n+1):
    harmonic_sum += 1/i
end_time = time.time()
print("Time Process  " + str(round(end_time - begin_time, 3)) + "s")
print(f"Harmonic sum {harmonic_sum} for n equals to {n}")

File: learning_cpp/exercices_from_claude_delannoy/chap_2/harmonic_sum.cpp
#include <iostream>
using namespace std;
#include <time.h>
#include <math.h>
int main()
{
    int n(pow(10,9));
    int i = 1;
    clock_t t;
    double harmonic(0);
    t = clock();
    for(i;i<=n;i++)
    {
        harmonic += (double)(1)/i;
    }
    t += -clock();
    double time_taken = double(t)*(-1.0) / CLOCKS_PER_SEC;
    cout << "Time taken : "<< time_taken << " in seconds\n"; 
    cout << "Harmonic sum " << harmonic << " for n equals to " << n << endl;
    return 0;
}